catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/TestTransactionRequestConverter.java

```diff
@@ -0,0 +1,150 @@
+
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+
+        MetacardImpl metacard = new MetacardImpl();
+        metacard.setId("metacard1");
+        InsertAction insertAction = new InsertAction("csw:Record", null, Arrays.asList(metacard));
+        transactionRequest.getInsertActions()
+                .add(insertAction);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+
+        String xml = xStream.toXML(transactionRequest);
+        Diff diff = XMLUnit.compareXML(xml, expectedInsertXML);
+        assertTrue(diff.similar());
+    }
+
+    @Test
+    public void testValidUpdateMarshal() throws SAXException, IOException, XpathException {
+
+        XStream xStream = new XStream(new Xpp3Driver());
+        xStream.registerConverter(new TransactionRequestConverter(cswRecordConverter));
+        xStream.alias("csw:" + CswConstants.TRANSACTION, CswTransactionRequest.class);
+
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+
+        MetacardImpl metacard = new MetacardImpl();
+        metacard.setId("metacard1");
+        UpdateAction updateAction = new UpdateAction(metacard, "csw:Record", null);
+        transactionRequest.getUpdateActions()
+                .add(updateAction);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+
+        String xml = xStream.toXML(transactionRequest);
+        Diff diff = XMLUnit.compareXML(xml, expectedUpdateXML);
+        assertTrue(diff.similar());
+    }
+
+    @Test
+    public void testValidDeleteMarshal() throws SAXException, IOException, XpathException {
+
+        XStream xStream = new XStream(new Xpp3Driver());
+        xStream.registerConverter(new TransactionRequestConverter(cswRecordConverter));
+        xStream.alias("csw:" + CswConstants.TRANSACTION, CswTransactionRequest.class);
+
+        CswTransactionRequest transactionRequest = new CswTransactionRequest();
+
+        MetacardImpl metacard = new MetacardImpl();
+        metacard.setId("metacard1");
+        DeleteType deleteType = new DeleteType();
+        QueryConstraintType queryConstraintType = new QueryConstraintType();
+        queryConstraintType.setCqlText("identifier = metacard1");
+        deleteType.setConstraint(queryConstraintType);
+        DeleteAction deleteAction = new DeleteAction(deleteType, null);
+        transactionRequest.getDeleteActions()
+                .add(deleteAction);
+        transactionRequest.setService(CswConstants.CSW);
+        transactionRequest.setVerbose(true);
+        transactionRequest.setVersion(CswConstants.VERSION_2_0_2);
+
+        String xml = xStream.toXML(transactionRequest);
+        Diff diff = XMLUnit.compareXML(xml, expectedDeleteXML);
+        assertTrue(diff.similar());
+    }
```

Comment:
Technically we support multiple Operations in a single request, we should have a test for it.